### ISSUE TITLE
Publish as FPWD

### ISSUE THREAD

**Issue by marcoscaceres** [w3c/geolocation-api#42]
The spec now contains some pretty significant updates since 2016:
 
 * Now Controlled by feature policy - https://github.com/w3c/geolocation-api/pull/41
 * Fixed confusion around callback interfaces not being an EventHandler https://github.com/w3c/geolocation-api/pull/39   
 * Only expose the API in SecureContexts  
 * Renamed the interfaces and dictionaries + removal of `NoInterfaceObject` + [SameObject] annotation on geolocation attribute  https://github.com/w3c/geolocation-api/pull/23
 
@plehegar, @anssiko, I think it's worth republishing new Recommendation. WDYT?

**Comment by anssiko** [comment]
>@plehegar, @anssiko, I think it's worth republishing new Recommendation. WDYT?

I agree. We have now addressed all the known open issues and the spec has been hardened for security and privacy in line with the group's maintenance commitment after taking the spec over from the now-defunct Geolocation WG. To my understanding, all the major implementations agree with the spec, but implementers please chime in if that's not the case.

@plehegar what is the path of least resistance (read: least make work) to publish a new Recommendation? Should we wait for the Process 2020 to be operational?

Cc @reillyeon for the co-chair perspective.

Cc @cdumez for WebKit.

**Comment by reillyeon** [comment]
I agree that a new Recommendation is warranted.

**Comment by marcoscaceres** [comment]
I don't mind going through CR again.

**Comment by xfq** [comment]
I agree with @plehegar that we should [publish a Candidate Recommendation with substantive changes (but no new features)](https://www.w3.org/Guide/transitions?profile=CR&cr=rec-amended), if the feature policy support is not considered a [new feature](https://www.w3.org/2019/Process-20190301/#correction-classes).

We also need wide review of the changes before the CR transition.

**Comment by marcoscaceres** [comment]
Ok, let me know if there is anything I can do to help!

**Comment by plehegar** [comment]
@anssiko , if you concur, it would be nice to have some kind of group decision to point to for this, and then @xfq can do the paper work.

**Comment by anssiko** [comment]
@plehegar [CfC to publish Geolocation API Amended Recommendation - review by 18 June 2020](https://lists.w3.org/Archives/Public/public-device-apis/2020Jun/0006.html)

**Comment by anssiko** [comment]
@xfq [CfC passed](https://lists.w3.org/Archives/Public/public-device-apis/2020Jun/0008.html), please feel free to proceed with the paper work.

**Comment by xfq** [comment]
Some information for myself:

New normative references since the last REC:

* `[FEATURE-POLICY]`
* `[HTML]`
* `[RFC8174]`
* `[secure-contexts]`
* `[url]`

Test suite: https://github.com/web-platform-tests/wpt/tree/master/geolocation-API

Preliminary implementation report: https://wpt.fyi/results/geolocation-API?label=experimental&label=master&aligned

CR exit criteria: two interoperable deployed implementations of each feature

Wide review: will file a GitHub issue

**Comment by marcoscaceres** [comment]
@xfq, #68 adds [Permissions API] as a dep.

**Comment by marcoscaceres** [comment]
Given that we are operating under the new charter + 2020 Process, I wonder if we should revisit the publication options? We are no longer blocked on using the 2020 Process, so...

**Comment by anssiko** [comment]
For discussion at  https://www.w3.org/events/meetings/a8a1448f-049b-468e-8b45-8eae6b8e75be#agenda

**Comment by marcoscaceres** [comment]
Awesome! thanks @anssiko. I'll chat to @plehegar in the meantime to get a better understanding of our options to bring to the meeting.

**Comment by marcoscaceres** [comment]
Hey Folks, @plehegar met this week to chat about the path forward. We looked at the requirements from the Process 2020 and, because we've added new features to the spec, we need to publish a FPWD... which turns out to be a great thing because it means we can significantly modernize the spec, then (relatively) quickly move it to be a "living standard".  

So, basically, this will be the plan: 

 * rewrite the spec, modernizing it to use Infra, etc. 
 * update the spec to deal with the visibility issue.

We can publish a FPWD whenever we like. Ideally we will do that straight after our meeting in a few weeks - which will give me time to make the above changes. After FPWD, @plehegar suggested we wait one month before going to CR. We can sit on CR until we get browser alignment on the page visibility issue (there are some web compat issues, so we need to decide which model is best and then figure out which engine needs updating). Alternatively, we move quickly to REC, then republish an updated REC to fix the visibility issue ~6 months down the line.  

If it's ok with everyone, I'd like to "officially" become Editor of this spec. I'd also be keen to have someone be a co-editor, to also edit or help review PRs (maybe someone new who hasn't edited before - preferably from an underrepresented community in our working group). Any taker to be my co-pilot?

**Comment by marcoscaceres** [comment]
In prep for FPWD: 

> must record the group’s decision to request advancement.

https://www.w3.org/2021/04/07-dap-minutes.html#r04

> must obtain Director approval.

@xfq, I can't recall the name of the repo where to file the request :( Could you assist me with this? 

> must publicly document all new features (class 4 changes) to the technical report since the previous publication.

https://w3c.github.io/geolocation-api/#changes-since-last-publication

> must publicly document if other substantive changes (class 3 changes) have been made, and should document the details of such changes.

https://w3c.github.io/geolocation-api/#changes-since-last-publication

> should publicly document if editorial changes have been made, and may document the details of such changes.

https://github.com/w3c/geolocation-api/commits/gh-pages

> must formally address all issues raised about the document since the previous maturity level.

All issues have been responded to in:
https://github.com/w3c/geolocation-api/issues

> must provide public documentation of any Formal Objections.

None. 

> should report which, if any, of the Working Group's requirements for this document have changed since the previous step.

None.

> should report any changes in dependencies with other groups.

WebAppSec for Permission API? 

> should provide information about implementations known to the Working Group.

Implementation in Blink, WebKit, and Gecko.

**Comment by xfq** [comment]
> @xfq, I can't recall the name of the repo where to file the request :( Could you assist me with this?
> 

It's https://github.com/w3c/transitions/issues and I am happy to help file the issue (actually I have created a [test FPWD snapshot](https://raw.githack.com/xfq/testing/f39cfca72fefbbab0ec8c1482373e7fc003e78a9/geolocation-WD/Overview.html)), but before filing the transition request, I found the following paragraph in our [charter](https://www.w3.org/2020/11/das-wg-charter.html#decisions):

> To afford asynchronous decisions and organizational deliberation, any resolution (including publication decisions) taken in a face-to-face meeting or teleconference will be considered provisional. A call for consensus (CfC) will be issued for all resolutions (for example, via email and/or web-based survey), with a response period from 5 to 10 working days, depending on the chair's evaluation of the group consensus on the issue. If no objections are raised on the mailing list by the end of the response period, the resolution will be considered to have consensus as a resolution of the Working Group.

It looks like we only have a [CfC](https://lists.w3.org/Archives/Public/public-device-apis/2020Jun/0006.html) for publishing this spec as an Amended Recommendation, but not for FPWD, so we probably need to issue a new CfC first...

**Comment by marcoscaceres** [comment]
> It looks like we only have a CfC for publishing this spec as an Amended Recommendation, but not for FPWD, so we probably need to issue a new CfC first...

Agree. @anssiko, do you want to "replay": https://lists.w3.org/Archives/Public/public-device-apis/2020Jun/0006.html

In the meantime, it would be great if someone from the group could step up and review the spec from start to finish. I basically rewrote the whole thing, so there are bound to be typos or potentially other issues.

**Comment by anssiko** [comment]
FYI: [CfC to publish Geolocation API First Public Working Draft - review by 26 April 2021](https://lists.w3.org/Archives/Public/public-device-apis/2021Apr/0011.html)

**Comment by anssiko** [comment]
The [CfC passed](https://lists.w3.org/Archives/Public/public-device-apis/2021Apr/0014.html). @xfq please proceed with the publication.

**Comment by xfq** [comment]
Since the shortname of the current REC is `geolocation-API` and an FPWD is considered a new document, we cannot reuse the same shortname, so the document has to be published under a different shortname.

We have two options:

* either go with `geolocation-API-2` and it's approved already
* or switch to `geolocation` but then, @plehegar needs to approve it

**Comment by marcoscaceres** [comment]
"geolocation" is my preference... versions/levels don't make any sense.

**Comment by reillyeon** [comment]
Having "geolocation-API" point to the REC and "geolocation" point to the WD sounds confusing. Are there no prior examples we can work from here? I have a slight preference for "geolocation-API-WD". I assume that when we publish another REC version of this specification it will overwrite the current document at "geolocation-API".

**Comment by marcoscaceres** [comment]
> geolocation-API-WD

Having that status in the short name is not a thing. 

How it will work is "geolocation-API" will become Superseded Recommendation, and "geolocation" will become the new canonical Recommendation.

**Comment by deniak** [comment]
I wouldn't recommend using `geolocation-API-WD` since it's the shortname that will be used to identify the specification even after it reaches CR.
Now, if the new document is published under the shortname `geolocation`, we could redirect https://www.w3.org/TR/geolocation-API/ to https://www.w3.org/TR/geolocation/.

**Comment by reillyeon** [comment]
> How it will work is "geolocation-API" will become Superseded Recommendation, and "geolocation" will become the new canonical Recommendation.

Does this mean that every time there is a new Recommendation the shortname needs to change or does this only apply because we're switching to a new Process?

**Comment by deniak** [comment]
> Does this mean that every time there is a new Recommendation the shortname needs to change or does this only apply because we're switching to a new Process?

It's actually a mix of things. With the previous process document, you can only revise a REC with non substantive changes. In this case, I understand you are adding new features. That's why @plehegar said you need to publish a FPWD.

The good news is the new process document facilitates that kind of [updates to existing RECs](https://www.w3.org/2020/Process-20200915/#revising-rec) so going forward, WGs won't need to publish a FPWD to add new features.

**Comment by anssiko** [comment]
Redirect from https://www.w3.org/TR/geolocation-API/ to https://www.w3.org/TR/geolocation/ solves the REC vs WD confusion so I believe we can settle on ”geolocation” as the shortname.

**Comment by xfq** [comment]
Published: https://www.w3.org/TR/2021/WD-geolocation-20210527/
